docs/Hardware-Agnostic-Training.md

```diff
@@ -0,0 +1,67 @@
+# Switching Hardware Mid Training Run
+
+One of the standout features of Levanter is its ability to seamlessly operate on both GPU and TPU platforms. This guide will walk you through the steps and best practices for switching hardware configurations mid-training, ensuring a smooth and efficient training process for your models.
+
+### Starting a Training Run
+
+After getting setup up in your [GPU](Getting-Started-GPU.md) or [TPU](Getting-Started-TPU-VM.md) environment (please see [our tutorials](Installation.md) on how to do so if you haven't already), the following command will kick off a training run.
+
+```bash
+python src/levanter/main/train_lm.py \
+    --config_path config/gpt2_small.yaml \
+    --trainer.checkpointer.base_path output/gpt2_small_webtext \
+    --trainer.train_batch_size 256 \
+    --trainer.per_device_parallelism 128
+```
+This example command assumes you're starting your training run on 2 GPUs in a [FSDP (Fully Sharded Data Parallel)](https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html) setup. The `trainer.per_device_parallelism` parameter instructs Levanter to distribute the processing of each training batch across multiple accelerators, in this case, across 2 GPUs. With this setting, each GPU handles a subset of the training batch, processing 128 examples each, and their combined efforts contribute to the effective handling of the entire 256 batch size.
```

Comment:
you can just leave out `per_device_parallelism` and it'll infer it to be 128. It'll simplify things